src/App.tsx

```diff
@@ -1,26 +1,266 @@
+        markings={marks} onClick={() => { this.toggleSelectSquare(i); }}/>;
+    }
+}
+
+interface SquareProps {
+    isPermanent: boolean;
+    isHighlighted: boolean;
+    isContradicting: boolean;
+    value: SquareValue | null;
+    markings: SquareValue[];
+    onClick: () => void;
+}
+
+type SquareValue = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 ;
+
+class Square extends React.Component<SquareProps, {}> {
+    public getMarks(): boolean[] {
+        const lst: boolean[] = [];
+        for (let i = 0; i < 9; i++) {
+            lst[i] = this.props.markings.includes(i as SquareValue);
+        }
+        return lst;
+    }
+    public render() {
+        const light: string = this.props.isHighlighted ? "highlight" : (this.props.isContradicting ? "contradict" : "");
+        const marks: boolean[] = this.getMarks();
+        const displayMarkings: (boolean | 0) = // determine if markings or value should be displayed
+            this.props.markings && this.props.markings.length &&
+            !this.props.isPermanent && !this.props.value;
+        if (displayMarkings) {
+            return(
+                <div className={"square " + light} onClick={this.props.onClick}>
+                    <div className="mark">{marks[1] ? 1 : null}</div>
+                    <div className="mark">{marks[2] ? 2 : null}</div>
+                    <div className="mark">{marks[3] ? 3 : null}</div>
+                    <div className="mark">{marks[4] ? 4 : null}</div>
+                    <div className="mark">{marks[5] ? 5 : null}</div>
+                    <div className="mark">{marks[6] ? 6 : null}</div>
+                    <div className="mark">{marks[7] ? 7 : null}</div>
+                    <div className="mark">{marks[8] ? 8 : null}</div>
+                    <div className="mark">{marks[9] ? 9 : null}</div>
+                </div>
+            );
+        } else {
+            return(
+                <div className={"square " + light} onClick={this.props.onClick}>{this.props.value}</div>
+            );
+        }
+    }
+}
+
+interface NumpadProps {
+    onClickNum: (i: SquareValue) => void;
+    onClickDel: () => void;
+}
+
+class Numpad extends React.Component<NumpadProps, {}> {
+    public renderNumButtons() {
+        const list = [];
+        for (let i = 1; i <= 9; i++) {
+            list.push(this.renderNumButton(i as SquareValue));
+        }
+        return list;
+    }
+
+    public renderNumButton(i: SquareValue) {
+        return <Button onClick={() => {this.props.onClickNum(i); }} className="button-num" variant="contained">{i as number}</Button>;
+    }
+
+    public render() {
+        return (
+            <div className="button-num-pad">
+                {this.renderNumButtons()}
+                <Button onClick={this.props.onClickDel} className="button" variant="contained">DELETE</Button>
+            </div>
+        );
+    }
+}
+
+class NormalButton extends React.Component<{}, {}> {
```

Comment:
Please inline these four classes.

Inline means don't create separate classes for each of these.